force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -27,9 +27,34 @@ public with sharing class NextStepFuture {
      * Opportunity Next Step field is a standard field
      * Update the corresponding test class AsynchronousTest and test method nextStepFuture_testUpdateAllNextStep to cover test this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    @future
+    public static void updateAllNextStep () {
+        // Retrieve all Accounts
+        List<Account> accsToUpdate = [SELECT Id, Next_Step__c
+                                        FROM Account
+                                        WHERE Next_Step__c = null
+                                        WITH SECURITY_ENFORCED];
+
+        // Retrieve all Opportunities
+        List<Opportunity> oppsToUpdate = [SELECT Id, NextStep
+                                            FROM Opportunity
+                                            WHERE NextStep = null
+                                            WITH SECURITY_ENFORCED];
+        
+        // Update the Next Step field on the Accounts
+        for (Account acc : accsToUpdate) {
+            acc.Next_Step__c = 'Meeting in the future';
+        }
+
+        // Update the Next Step field on the Opportunities
+        for (Opportunity opp : oppsToUpdate) {
+            opp.NextStep = 'Meeting in the future';
+        }
+
+        // Update the Accounts and Opportunities
+        Database.update(accsToUpdate, AccessLevel.USER_MODE);
+        Database.update(oppsToUpdate, AccessLevel.USER_MODE);
```

Comment:
Good work. Another solution here would be to use SOQL for loop or an inner query. And you can assign your records to a list o SOBjects and do one DML rather than two. 
```
 public static void setNextStep(){
        List<SObject> recordsToUpdate = new List<SObject>();
       String meetingStatus = 'Meeting in the future';
        for(Account account : [SELECT Next_Step__c, (SELECT NextStep FROM Opportunities)
        FROM Account
        WITH SYSTEM_MODE ]){
            account.Next_Step__c = meetingStatus;
            recordsToUpdate.add(account);
           for(Opportunity opportunity : account.Opportunities){
               opportunity.NextStep = meetingStatus;
               recordsToUpdate.add(opportunity);
           }
        }
        Database.update(recordsToUpdate, AccessLevel.SYSTEM_MODE);
    }
    ```

Reply:
@bdJohnson72 I actually played around with this one for a while and could not get it to work with the inner query. I think you just solved my problem I was assigning it to a List<Account> instead of List<SObject>. Thank you!